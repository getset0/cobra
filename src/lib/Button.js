import styled from "styled-components";

const Button = styled.button`
  background-color: palevioletred
  padding: ${props => props.small ? "0.25em 1em" : "0.5em 2em"};
`;

export default Button;
