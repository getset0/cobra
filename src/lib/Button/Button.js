import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { theme } from "../constants";

const Button = styled.button`
  line-height: 40px;
  border-radius: 5px;
  height: 40px;
  width: auto;
  font-weight: 600;
  letter-spacing: 1px;
  text-decoration: none;
  white-space: nowrap;
  font-size: 1em;
  appearance: none;
  padding: 0em 2em;
  background: transparent;
  margin: 0 2px 15px;
  outline: 0;
  transition: all .1s ease-out;

  ${props => !props.disabled && css`
    &:hover {
      opacity: 0.9;
      cursor: pointer;
      transform: scale(1.02);
    }

    &:active {
        opacity: 1;
        transform: scale(1);
    }
  `}

  ${props => props.small && css`
    height: 30px;
    line-height: 30px;
    padding: 0em 1em;
    font-size: 0.85em;
  `}

  ${props => props.large && css`
    height: 48px;
    line-height: 48px;
    padding: 0em 2.5em;
    font-size: 1.15em;
  `}

  color: ${props => props.theme.primaryColor};
  border: 2px solid ${props => props.theme.primaryColor};

  ${props => props.uppercase && css`
    text-transform: uppercase;
  `}

  ${props => props.primary && css`
    background: ${props => props.theme.primaryColor};
    color: ${props => props.theme.whiteColor};
  `}

  ${props => props.secondary && css`
    border: 2px solid ${props => props.theme.secondaryColor};
    background: ${props => props.theme.secondaryColor};
    color: ${props => props.theme.whiteColor};
  `}

  ${props => props.disabled && css`
    opacity: 0.5;
    &:hover {
      cursor: not-allowed;
    }
  `}
`;

Button.propTypes = {
  disabled: PropTypes.bool,
  secondary: PropTypes.bool,
  primary: PropTypes.bool,
  uppercase: PropTypes.bool,
  large: PropTypes.bool,
  small: PropTypes.bool,
  children: PropTypes.any.isRequired,
  theme: PropTypes.object
};

Button.defaultProps = {
  disabled: false,
  secondary: false,
  primary: false,
  uppercase: false,
  large: false,
  small: false,
  theme: theme
};

export default Button;
