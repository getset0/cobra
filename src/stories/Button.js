import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withBackgrounds } from "@storybook/addon-backgrounds";
import { backgroundOptions } from "./util";
import { Button } from "../lib";

storiesOf("Button", module)
  .addDecorator(withInfo)
  .addDecorator(withBackgrounds(backgroundOptions))
  .add("Normal", () => (
    <React.Fragment>
      <Button small>Small Button</Button>
      <Button>Normal Button</Button>
      <Button large>Large Button</Button>
      <Button uppercase>Normal Uppercase Button</Button>
    </React.Fragment>
  ))
  .add("Primary", () => (
    <React.Fragment>
      <Button primary small>
      Small Button
      </Button>
      <Button primary>Primary Button</Button>
      <Button primary large>
      Large Button
      </Button>
      <Button primary uppercase>
      Primary Button
      </Button>
    </React.Fragment>
  ))
  .add("Secondary", () => (
    <React.Fragment>
      <Button secondary small>
      Small Button
      </Button>
      <Button secondary>Secondary Button</Button>
      <Button secondary large>
      Large Button
      </Button>
      <Button secondary uppercase>
      Secondary Button
      </Button>
    </React.Fragment>
  ))
  .add("Disabled", () => (
    <React.Fragment>
      <Button disabled>Disabled Button</Button>
      <Button disabled uppercase>
      Disabled Button
      </Button>
    </React.Fragment>
  ));
