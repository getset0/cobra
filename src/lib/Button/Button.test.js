import React from "react";
import Button from "./Button";
import "jest-styled-components";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

it("renders button without crashing", () => {
  shallow(<Button>Test</Button>);
});

it("renders normal button correctly", () => {
  const wrapper = shallow(<Button>Test</Button>);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
