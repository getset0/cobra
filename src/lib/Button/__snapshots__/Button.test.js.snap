// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders normal button correctly 1`] = `
<StyledComponent
  disabled={false}
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  border-radius: 5px;
  font-weight: 600;
  letter-spacing: 1px;
  text-decoration: none;
  white-space: nowrap;
  font-size: 1rem;
  appearance: none;
  padding: 1em 2em;
  background: transparent;
  margin: 0 2px 15px;
  outline: 0;
  transition: transform 100ms ease-out, box-shadow 100ms ease-out;

  ",
          [Function],
          "

  ",
          [Function],
          "

  ",
          [Function],
          "

  color: ",
          [Function],
          ";
  border: 2px solid ",
          [Function],
          ";

  ",
          [Function],
          "

  ",
          [Function],
          "

  ",
          [Function],
          "

  ",
          [Function],
          "
",
        ],
      },
      "defaultProps": Object {
        "disabled": false,
        "large": false,
        "primary": false,
        "secondary": false,
        "small": false,
        "theme": Object {
          "blackColor": "#000000",
          "darkBlueColor": "#02469B",
          "darkGrayColor": "#4A4A4A",
          "darkestBlueColor": "#02254D",
          "darkestGrayColor": "#303030",
          "greenColor": "#76B041",
          "lightBlueColor": "#0190D0",
          "lightGrayColor": "#D8D8D8",
          "lightestBlueColor": "#00FEFE",
          "mediumGrayColor": "#979797",
          "primaryColor": "#1676B7",
          "secondaryColor": "#17BEBB",
          "whiteBackgroundColor": "#fdfdfd",
          "whiteColor": "#ffffff",
          "yellowColor": "#FFC914",
        },
        "uppercase": false,
      },
      "displayName": "styled.button",
      "propTypes": Object {
        "children": [Function],
        "disabled": [Function],
        "large": [Function],
        "primary": [Function],
        "secondary": [Function],
        "small": [Function],
        "theme": [Function],
        "uppercase": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  large={false}
  primary={false}
  secondary={false}
  small={false}
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
  uppercase={false}
>
  Test
</StyledComponent>
`;
